include/aie/Dialect/AIEVec/IR/AIEVecTypeConstraints.td

```diff
@@ -146,4 +146,17 @@ class IsValidAIE2MatMulShapeAndType<string lhs, string rhs, string acc> :
                                    rhs, VectorOfShapeAndType<[8, 4], BF16>,
                                    acc, VectorOfShapeAndType<[4, 4], F32>>]>>;
 
+class hasAnyTypeOf<string name, list<Type> vaildTypes> :
+  SubstLeaves<"$_self", ElementType<name>.result, AnyTypeOf<vaildTypes>.predicate>;
```

Comment:
An equivalent predicate exists upstream already.

Reply:
I replaced it with an existing predicate `ElementTypeIsPred`.